src/api/flaskr/service/learn/pdf_export.py

```
@@ -0,0 +1,1124 @@
+    html_path.write_text(content, encoding="utf-8")
+
+    repo_root, script_path = _resolve_pdf_html_renderer_script()
+    command = [
+        "node",
+        str(script_path),
+        str(html_path),
+        str(png_path),
+        "1280",
+    ]
+    try:
+        subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)